### DISCUSSION TITLE
uPET in an original PET 2001-8 case

### DISCUSSION THREAD

**Discussion by Sch-LikA** [fachat/MicroPET#11] [Q&A] [answered]
I just bought an original PET-2001-8 machine, a complete one only missing the mainboard. It has the original "chiclet" keyboard, and even the original cassette player :-)

I would love building an 'adapter' PCB that would sit in the case, with the connectors in the right spots like the original mainboard, able to get the uPET to give life back to that old case.

So my questions :

- Can the original "Chiclet" keyboard be somehow used with the uPET ? According to the Github page, it seems only the Graphics keyboard is supported, maybe needing a different matrix in the rom ? I may add a connector for an additional external keyboard as it seems the chiclet keyboard is a pain to use
- Can the video board drive the original PET monitor ? I think it only needs luma + sync. Would it need a custom video output circuit ?
- Is there any other point I am missing ?

**Comment by jacgoudsmit** [comment]
I think the thing you're looking for already exists: check out the MiniPet by Tynemouth at https://www.thefuturewas8bit.com/shop/tynemouth-products/minipet4080.html

They're working on an updated version but I have the first version and that was designed as a drop-in replacement for PET motherboards. It even allowed you to put it on existing standoffs in the case and have the card-edge connectors show up on the back side.

It looks like most pictures show their PET keyboard but it may still be possible to buy one that doesn't have the keyboard.

EDIT: I just found out that David Curran at Tynemouth Software who designed the MiniPet, has taken his shop offline because of mounting problems with shipping. That's too bad. I hope he'll be back.

**Comment by Sch-LikA** [comment]
Well I know about it but I really prefer microPET's design and features, that's why.

**Comment by fachat** [comment]
I have thought about such a board as well. My Eagle PCB design software is limited to Euro PCB size though (old non-profit version before they switched to subscription-based licenses), so I can't so such a board.
Steve Gray had a go at converting the schematics to keycad and IIRC was successful, so it should be possible.

However, with a larger board, I'd actually include color PET capability, and even some expansion slots...

**Reply by Sch-LikA** [reply]
Wow, would be the ultimate PET for sure. Any link to the Kicad (I suppose) version ?

Anyway, with current R2 design, can the original "chiclet" keyboard be used and can the microPET drive the PET's original CRT ?

Thanks !

**Comment by fachat** [comment, marked as answer]
The chicklet keyboard has AFAIK the same matrix table as the N-type keyboard, so should work.
Drinving the original CRT ... that would mean to resurrect the Composite video timing option from R1, that I dropped from R2. Doable I suppose, maybe with slower speed then.
As for the kicad version ... I don't think Steve has finished it, but he is reachable on github as well -> @sjgray https://github.com/sjgray

Edit: the "composite" option has of course the three needed signals video, hsync and vsync available also. It's just that the "composite" timing is similar to the PET's timing.

**Reply by Sch-LikA** [reply]
Thanks for the precisions !
